downloads/swan-lake-release-notes/swan-lake-2201.7.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,249 @@
+
+## Language updates
+
+### New features
+
+#### Introduction of the `group by` and `collect` clauses
+
+The language now supports the `group by` and `collect` clauses to perform aggregation-related operations.
+
+The `group by` clause is used to group a collection based on a grouping-key. The grouping-key will be unique to each group.
+
+```ballerina
+import ballerina/io;
+
+type Order record {|
+    string name;
+    float price;
+|};
+
+public function main() {
+    Order[] orders = [{name: "Rich Dad Poor Dad", price: 16.0}, {name: "Becoming", price: 22.5}, {name: "Rich Dad Poor Dad", price: 16.4}, {name: "Becoming", price: 22.6}];
+    var averages = from var {name, price} in orders
+        group by name
+        select {name, avg: avg(price)};
+    io:println(averages); // [{"name":"Rich Dad Poor Dad","avg":16.2},{"name":"Becoming","avg":22.55}]
+}
+```
+
+The `collect` clause is used to group a collection into one group.
+
+```ballerina
+import ballerina/io;
+
+type Order record {|
+    string name;
+    float price;
+|};
+
+public function main() {
+    Order[] orders = [
+        {name: "Rich Dad Poor Dad", price: 16.0},
+        {name: "Becoming", price: 22.6},
+        {name: "Rich Dad Poor Dad", price: 16.4},
+        {name: "Becoming", price: 22.5}
+    ];
+    var average = from var {price} in orders
+        collect avg(price);
+    io:println(average); // 19.375
+}
+```
+
+### Improvements
+
+#### Support for constant declarations without a type descriptor
+
+Constants can now be declared without specifying a type descriptor. Previously, it was required to specify a type descriptor except with numeric and string literal expressions.
+
+Both of the following cases are allowed now.
+
+```ballerina
+const map<string> HTTP_OK = {httpCode: "200", message: "OK"};
+const HTTP_BAD_REQUEST = {httpCode: "400", message: "BAD REQUEST"};
+```
+
+#### Support for unions as type descriptors in constant declarations
+  
+Constant declarations with unions as type descriptors are now supported.
+
+```ballerina
+const float|decimal TOTAL = 1 + 2.0;
+```
+
+#### Support for more recursive type definitions
+
+The type definition resolution logic has been improved to support more recursive type definitions.
+
+```ballerina
+type A [A];
+type B [B] & readonly;
+```
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for Swan Lake 2201.7.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+label%3ATeam%2FCompilerFE+milestone%3A2201.7.0+is%3Aclosed+label%3AType%2FBug).
+
+## Runtime updates
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.7.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.7.0+label%3ATeam%2FjBallerina+label%3AType%2FBug+is%3Aclosed).
```

Comment:
Why using two different ways for link text. Compare this and line 174.

Reply:
Fixed.